project/driver_controls/src/input_interrupt.c

```diff
@@ -0,0 +1,149 @@
+#include "adc.h"
+#include "input_interrupt.h"
+#include "event_queue.h"
+#include "stm32f0xx.h"
+
+#define NUM_INPUT_EVENTS 16
+
+#define COAST_THRESHOLD 1000
+#define DRIVE_THRESHOLD 3000
+
+typedef void (*DriverInput)(GPIOAddress address, Event *e);
+
+typedef struct AnalogStatus {
+  InputEvent pedal;
+} AnalogStatus;
```

Comment:
I kept it as a struct so that it would be easy to include other analog inputs when they need to be added.